File: src/Shared/Munchkin.Core/Cards/Card.cs
namespace Munchkin.Core.Cards;

public class Card
{
    public string Name { get; }

    public CardTypes Type { get; }

    public int Power { get; }

    public Card(string name, int power, CardTypes type)
    {
        Name = name;
        Power = power;
        Type = type;
    }
}

public enum CardTypes
{
    Equipment,
    Monster
}

File: frontend/src/api/Cards/Card.ts
export type Card =
{

}

export enum CardBackType
{
    Treasure,
    Door
}

export enum CardTypes
{
    Monster,
    Curse,
    Clothes,
    Improvement
}

export function getBadgeImage(type: CardTypes) : string
{
    switch (type)
    {
        case CardTypes.Curse: return "assets/cards/types/curse.png"
        case CardTypes.Improvement: return "assets/cards/types/improvement.png"
        case CardTypes.Clothes: return "assets/cards/types/clothes.png"
        case CardTypes.Monster: return "assets/cards/types/monster.png"
    }
}